presto-main/src/main/java/com/facebook/presto/split/SplitManager.java

```diff
@@ -54,20 +63,30 @@ public void removeConnectorSplitManager(ConnectorId connectorId)
         splitManagers.remove(connectorId);
     }
 
-    public SplitSource getSplits(Session session, TableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    public SplitSource getSplits(Session session, TableHandle table, SplitSchedulingStrategy splitSchedulingStrategy)
     {
-        ConnectorId connectorId = layout.getConnectorId();
+        ConnectorId connectorId = table.getConnectorId();
         ConnectorSplitManager splitManager = getConnectorSplitManager(connectorId);
 
         ConnectorSession connectorSession = session.toConnectorSession(connectorId);
+        // TODO add support in metadata to check Connector still needs TableLayoutHandle and method in ConnectorSplitManager to use connectorTableHandle to get split
+        ConnectorTableLayoutHandle layout;
+        if (!table.getLayout().isPresent()) {
+            Optional<TableLayoutResult> result = metadata.getLayout(session, table, Constraint.alwaysTrue(), Optional.empty());
+            checkArgument(result.isPresent(), format("must be able to obtain a layout from this connector %s", connectorId));
```

Comment:
nit:  For error message, let just say "unable to get table layout from connector".

Also, use string concat instead of format. See "Code Style" (https://github.com/prestodb/presto/blob/master/README.md#code-style) : 

> ... Sometimes, if you only need to append something, consider using the + operator.